app/src/main/java/vn/sunasterisk/music_67/ui/detailplaying/DetailPlayingActivity.kt

```diff
@@ -110,11 +129,50 @@ class DetailPlayingActivity : AppCompatActivity(), View.OnClickListener, TrackSt
 		currentTrack()
 	}
 
+	override fun startListener() {
+		createRotation()
+		objectAnimator.start()
+	}
+
 	private fun registerListener() {
 		imagePlay.setOnClickListener(this)
 		imageNext.setOnClickListener(this)
 		imagePrevious.setOnClickListener(this)
 		seekBarPlaying.setOnSeekBarChangeListener(this)
+		imageShuffle.setOnClickListener(this)
+		imageLoop.setOnClickListener(this)
+	}
+
+	private fun handleShuffle() {
+		when (countShuffleClick % 2) {
```

Comment:
Thực ra nó chỉ có 2 trạng thái thôi nên em không cần count đâu. Em chỉ cần từ service gọi ra thằng manager lấy shuffletype xem nếu là off em set là on nếu là on em set là off. tương tự vs update image

Reply:
cái này là khi user click vào shuffle on of thì báo về service r service báo cho manager ý a